example/Makefile

```diff
@@ -1,30 +1,49 @@
+# It will be == kb / new layout less the targets to work with GO
+# see that we cannot add the generate (ansible/helm) will not gen code
+# go test, vet fmt ^
 
 # Image URL to use all building/pushing image targets
-IMG ?= quay.io/joelanford/nginx-operator
+IMG ?= controller:latest
+# Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
+CRD_OPTIONS ?= "crd:trivialVersions=true"
 
-all: docker-build
+# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
+else
+GOBIN=$(shell go env GOBIN)
+endif
+
+all: manager
+
+# Build manager binary
+manager: manifests docker-build
 
 # Run against the configured Kubernetes cluster in ~/.kube/config
-run:
+run: manifests
 	go run ../main.go
 
 # Install CRDs into a cluster
-install:
-	kustomize build config/crd | kubectl apply -f -
+install: manifests kustomize
+	$(KUSTOMIZE) build config/crd | kubectl apply -f -
 
 # Uninstall CRDs from a cluster
-uninstall:
-	kustomize build config/crd | kubectl delete -f -
+uninstall: manifests kustomize
+	$(KUSTOMIZE) build config/crd | kubectl delete -f -
 
 # Deploy controller in the configured Kubernetes cluster in ~/.kube/config
-deploy:
-	cd config/manager && kustomize edit set image controller=${IMG}
-	kustomize build config/default | kubectl apply -f -
+deploy: manifests kustomize
+	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
+	$(KUSTOMIZE) build config/default | kubectl apply -f -
 
+# It is something that we should push to upstream since is valid for all scenarios
 # Undeploy controller in the configured Kubernetes cluster in ~/.kube/config
 undeploy:
-	cd config/manager && kustomize edit set image controller=${IMG}
-	kustomize build config/default | kubectl delete -f -
+	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG}
+	$(KUSTOMIZE) build config/default | kubectl delete -f -
+
```

Comment:
I think that It is something that we should push to upstream since is valid for all scenarios

Reply:
Done. Pushed to upstream a PR with. 
We also use it in the kb samples.